[export] fix joint graph metadata

pianpwk opened the pull request:
Differential Revision: D62652832

tugsbayasgalan commented on torch/_functorch/aot_autograd.py:
```diff
@@ -1301,7 +1301,8 @@ def flattened_joint(*args):
             return *fw_outs, *output_gradients
 
-        fx_g = make_fx(flattened_joint)(*full_args)
+        flattened_joint._orig_mod = fx_g
```
Can we use functools.wraps to get the orig mod? Can you also reference this note here https://github.com/pytorch/pytorch/pull/115454/files#diff-594ef3e84a8e3555bfa4a6167383949da5f321e93f602246eb9c0a797d5ecc00R617 
cc: @bdhirsh

pianpwk commented on test/export/test_experimental.py:
```diff
@@ -327,6 +328,40 @@ def forward(self, x):
+        inputs = (x, labels)
+
+        ep = export(net, inputs)
+        ep = _export_forward_backward(ep)
```
@bdhirsh it crashes at the ExportedProgram verifier, as nn_module_stack and other metadata is missing for the graph. 

I think it affects this line, where ModuleStackTracer populates the metadata: https://github.com/pytorch/pytorch/blob/4d3c0fc0618c9ae9b330d9efdb66623617919326/torch/fx/experimental/proxy_tensor.py#L1872-L1876

pianpwk commented:
updated to use functool.wraps on flattened_joint, let's see if this breaks anything

pianpwk committed:
[export] fix joint graph metadata (#136011)

Summary:
Fixes up missing metadata on export -> joint graph export, with patching in AOTAutograd so tracer knows to record nn_module_stack

X-link: https://github.com/pytorch/executorch/pull/5447


Test Plan: test_experimental

Reviewed By: tugsbayasgalan

Differential Revision: D62652832
